providers/mlx5/qp.c

```diff
@@ -942,8 +942,7 @@ static inline int _mlx5_post_send(struct ibv_qp *ibqp, struct ibv_send_wr *wr,
 			mlx5_bf_copy(bf->reg + bf->offset, (unsigned long long *)ctrl,
 				     align(size * 16, 64), qp);
 		else
-			mlx5_write64((__be32 *)ctrl, bf->reg + bf->offset,
-				     &ctx->lock32);
+			mmio_write64_be(bf->reg + bf->offset, *(__be64 *)ctrl);
```

Comment:
Didn't you miss here the endian handling that was previously done as part of mlx5_write64() ->MLX5_PAIR_TO_64 ? looks like mmio_write64_be just store the value.